Update version to 7.0.0-alpha1

rjernst opened the pull request:
This commit updates the version for master to 7.0.0-alpha1. It also adds
the 6.1 version constant, and fixes many tests, as well as marking some
as awaits fix.

rjernst committed:
Update version to 7.0.0-alpha1

This commit updates the version for master to 7.0.0-alpha1. It also adds
the 6.1 version constant, and fixes many tests, as well as marking some
as awaits fix.

rjernst committed:
fix precommit

rjernst committed:
more test fixes

rjernst committed:
more test fixes

rjernst committed:
Merge branch 'master' into bump7

rjernst commented:
I just pushed this branch to elastic/elasticsearch, hopefully someone can take this over as I am out the next 2 days and there are still lingering tests.

s1monw commented:
@rjernst I don't see any branch on `elastic/elasticsearch` is it safe to work off your branch?

s1monw committed:
Merge branch 'master' into bump7

rjernst commented:
Yes please do. Sorry I thought I had pushed it.

s1monw committed:
make sure default analyzer is always referenced by name [default]

s1monw committed:
Fix expected bytes in tests due to new major version

In 6.x we had to serialize the 2 byte marker for the transport address type
for BWC. Now in 6.x this doesn't exist and we write 2 bytes less when we serialize
a transport address to a min compat node.

Closes #25893

s1monw committed:
add back exception when spcifying broken setting for defautl analyzer

s1monw committed:
Pass in explicit version

Closes #25870

s1monw committed:
fix translog tests

ywelsch commented on core/src/main/java/org/elasticsearch/index/mapper/AllFieldMapper.java:
```diff
@@ -30,6 +30,7 @@
 import org.elasticsearch.common.lucene.all.AllTermQuery;
 import org.elasticsearch.common.settings.Settings;
 import org.elasticsearch.common.xcontent.XContentBuilder;
+import org.elasticsearch.common.xcontent.json.JsonXContent;
```
revert this? I don't see it being used.

ywelsch commented on core/src/test/java/org/elasticsearch/cluster/serialization/ClusterSerializationTests.java:
```diff
@@ -139,7 +138,7 @@ public void testSnapshotDeletionsInProgressSerialization() throws Exception {
         // serialize with old version
         outStream = new BytesStreamOutput();
-        outStream.setVersion(Version.CURRENT.minimumIndexCompatibilityVersion());
+        outStream.setVersion(Version.V_5_0_0);
```
Why V_5_0_0? we should never serialize to such a node?

s1monw replied:
yeah so I think this test can go away but I didn't wanna do it in this PR.

martijnvg reviewed (approved):
Looks good as far as I can see.

s1monw committed:
fix test

s1monw committed:
revert unnecessary import

s1monw committed:
fix IndexingIT

s1monw committed:
Merge branch 'master' into bump7

nik9000 commented on build.gradle:
```diff
@@ -79,18 +79,20 @@ int lastPrevMinor = -1 // the minor version number from the prev major we most r
-    if (currentVersion != foundVersion) {
+    if (currentVersion != foundVersion
+        && (major == prevMajor || major == currentVersion.major)
+        && (versions.isEmpty() || versions.last() != foundVersion)) {
```
I *think* this adds all the betas and rcs are wire and index compatible versions. I think, at least in the end, we want them to only be index and wire compatible if we only have only have them. Otherwise we'll be testing compatibility against betas while we're well into 6.0. Is this a thing we should handle in a followup?

nik9000 commented on qa/full-cluster-restart/src/test/java/org/elasticsearch/upgrades/FullClusterRestartIT.java:
```diff
@@ -62,7 +63,7 @@
     private final boolean runningAgainstOldCluster = Booleans.parseBoolean(System.getProperty("tests.is_old_cluster"));
     private final Version oldClusterVersion = Version.fromString(System.getProperty("tests.old_cluster_version"));
-    private final boolean supportsLenientBooleans = oldClusterVersion.onOrAfter(Version.V_6_0_0_alpha1);
+    private final boolean supportsLenientBooleans = oldClusterVersion.before(Version.V_6_0_0_alpha1);
```
+1

nik9000 commented on qa/mixed-cluster/src/test/java/org/elasticsearch/backwards/IndexingIT.java:
```diff
@@ -213,19 +213,20 @@ public void testSeqNoCheckpoints() throws Exception {
             int numDocs = 0;
             final int numberOfInitialDocs = 1 + randomInt(5);
             logger.info("indexing [{}] docs initially", numberOfInitialDocs);
-            numDocs += indexDocs(index, 0, numberOfInitialDocs);
```
I'd like someone else to have a look at this file's changes.

s1monw replied:
@ywelsch did look at it

nik9000 replied:
:+1:

nik9000 commented on test/framework/src/test/java/org/elasticsearch/test/VersionUtilsTests.java:
```diff
@@ -147,24 +149,22 @@ public void testResolveReleasedVersionsForStableBtranchBehindStableBranch() {
         public static final Version V_6_0_0_alpha1 = Version.fromString("6.0.0-alpha1");
         public static final Version V_6_0_0_alpha2 = Version.fromString("6.0.0-alpha2");
         public static final Version V_6_0_0_beta1 = Version.fromString("6.0.0-beta1");
-        public static final Version CURRENT = V_6_0_0_beta1;
```
The intent of these tests was to not have to change them when we upgraded unless we found a new kind of layout. So maybe we don't have to change this?

s1monw replied:
not sure what you mean here?

nik9000 replied:
I'm asking if you have to change this test at all. When I wrote the test I used all the `Version.fromString` so that you wouldn't have to change the tests as the versions marched along. I intended for the test to stay the same rather than mirror the current branch.

s1monw replied:
got it - I fixed it

nik9000 commented on test/framework/src/test/java/org/elasticsearch/test/VersionUtilsTests.java:
```diff
@@ -174,37 +174,29 @@ public void testResolveReleasedVersionsForUnstableBranch() {
-        // Because things blow up all over the place if the minimum compatible version isn't released.
-        // We'll fix this very, very soon. But for now, this hack.
-        // end big horrible hack
+        Version minimumCompatibleVersion = Version.CURRENT.minimumCompatibilityVersion();
```
Awesome.

nik9000 reviewed (commented):
Mostly LGTM. I left a few minors that can be followups and I've marked one file as beyond my ability to properly review so I'd like another set of eyes on it.

s1monw committed:
remove fixme

s1monw committed:
fix test to remove old version that area not relevant

s1monw commented:
@rjernst I think we are ready here. I still have some unaddressed comments from @nik9000 but it should be ready for you to merge when you are back!

imotov commented:
@s1monw if you bump the version today, could you also bump the version [here](https://github.com/elastic/elasticsearch/blob/aa007c1056be793062129495d5ce47f36c59f4cd/core/src/main/java/org/elasticsearch/cluster/SnapshotsInProgress.java#L419) and [here](https://github.com/elastic/elasticsearch/blob/aa007c1056be793062129495d5ce47f36c59f4cd/core/src/main/java/org/elasticsearch/cluster/SnapshotsInProgress.java#L463). I didn't want to backported it before running tests for a bit. So, it is currently only merged into master and if you start running 7.0.0 upgrade from 6.0.0 test, it might fail since the 6.0 branch doesn't have it. I will backport everything on Monday or over the weekend and change the version appropriately.

s1monw committed:
Merge branch 'master' into bump7

s1monw committed:
bump version in SnapshotInProgress

s1monw commented:
@imotov I merged master into this and bumped the version.

s1monw committed:
revert unnecessary changes

s1monw reviewed (approved):
LGTM

rjernst committed:
Merge branch 'master' into bump7

elasticmachine commented:
Pinging @elastic/es-core-infra
